[pangyyen] iP

pangyyen opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." – David Allen

DukePro frees your mind of having to remember things you need to do. It's:

- Text-based
- Easy to learn
- ~~FAST~~, SUPER FAST to use

All you need to do is:

1. Double-click it.
2. Add your tasks.
3. Let it manage your tasks for you 😉

And it is **FREE**!

## Features

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a _Java programmer_, you can use it to practice Java too. Here's the `main` method:

```java
  public static void main(String[] args) {
      Duke duke = new Duke();
      duke.start();
  }
```

Please review the [full PR here](https://github.com/nus-cs2103-AY2324S1/ip/pull/268)  for more details and let us know your feedback.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

pangyyen committed:
Rename, Greet, Exit

(a) Give your chatbot a new name
(b)  Implement an initial skeletal version

pangyyen committed:
Echo

echos commands entered by the user, and exits when the user types the command bye

pangyyen committed:
level-2 Add List

create Chatbot and TaskList

pangyyen committed:
level-3 Mark as Done

it could unmark as not done as well

pangyyen committed:
level-4 ToDos, Events, Deadlines

pangyyen committed:
A-TextUiTesting

Automated Text UI Testing

pangyyen committed:
level-5 handle errors

DukeException is created, and extended to DeadlineException, EventException

pangyyen committed:
level-6 Delete

TaskList is modified as dynamic list which allows dynamic delete, deleteException is also introduced

pangyyen committed:
level-7 Save

pangyyen committed:
Add javadoc

pangyyen committed:
Merge branch 'branch-level-7'

Javadoc are added to the newly created public method

pangyyen committed:
Change date format from String to java.time.LocalDate

pangyyen committed:
level-8 Fix format for input and output date

pangyyen committed:
Exstraction of user-input handling out of main method 'start()'

pangyyen committed:
Merge branch 'branch-level-8'

Handle date as java.time.LocalDate instead of String

pangyyen committed:
Move Chatbot to Duke

pangyyen committed:
Clear error in Command.java and Parser.java

pangyyen committed:
Clear all existing errors in Storage.java

pangyyen committed:
A-MoreOOP: abstract Ui, Storage, Parser

pangyyen committed:
Merge branch 'branch-A-MoreOOP'

Abstract out Ui, Storage and Parser class

pangyyen committed:
A-Packages: Organize files into Packages

pangyyen committed:
Merge branch 'branch-A-Packages'

Organize the java files into duke, duke.commands, duke.task, duke.storage, duke.ui, duke.exceptions"

pangyyen committed:
Move Duke.java into package duke.main

pangyyen committed:
Merge branch 'branch'

Add add-gradle-support to project directory

pangyyen committed:
Build Gradle

pangyyen committed:
Merge branch 'branch-A-Gradle'

Build gradle and set mainClass to duke.main.Duke

pangyyen committed:
Build JUnit tests

pangyyen committed:
Merge branch 'branch-A-JUnit'

Build JUnit test for commands, Task and Ui packages

pangyyen committed:
branch-A-JAR

pangyyen committed:
Merge branch 'branch-A-JAR'

Create JAR file for Duke

pangyyen committed:
Javadoc

pangyyen committed:
Merge branch 'branch-A-JavaDoc'

Add in javadoc for non-trivial methods

pangyyen committed:
Add checkstyles

pangyyen committed:
Merge branch 'branch-CodingStandard'

Add checkstyle in Gradle

pangyyen committed:
find tasks in list

pangyyen committed:
Merge branch 'branch-find'

Create find function to look for task containing certain description

thaddeusong commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,65 @@
+        this.commandType = commandType;
+    }
+
+    public Command(String commandType, String description) {
```
Rather than creating all these different public constructors, would it be easier to write an Of factory method and change these constructors to private?

thaddeusong reviewed (commented):
Couldn't find any coding standard issues, well done!

pangyyen committed:
Fix Duke.java and Exception classes styling errors

pangyyen committed:
Fix stlying errors in commands, task packages

pangyyen committed:
Fix all checkstyles error in Test Packages

pangyyen committed:
Add default JavaFx UI

pangyyen committed:
Combine Main.java into Launcher.java

pangyyen committed:
Debug showMessage() in Ui, make it a non-static method

pangyyen committed:
Integrate Duke into JavaFX, without fuke loading and saving

pangyyen committed:
Incorparate auto-termination javaFX by bye

pangyyen committed:
Add in file reading and saving

pangyyen committed:
Check style

pangyyen committed:
Merge branch 'branch-level-10'

Incorporate JavaFX as UI

wjacobw commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -112,6 +112,22 @@ public void markTaskAsNotDone(int taskListIndex) {
     }
 
+    /**
+     * Searches for tasks containing the given keyword and returns a list of matching tasks.
```
Good job on writing comments targeting other programmers reading the code.

wjacobw commented on src/main/java/duke/main/Duke.java:
```diff
@@ -156,6 +157,27 @@ private void saveTasksToFile(TaskList taskList) {
+     *
+     * @param keyword The keyword to search for.
+     */
+    public void findTasksByKeyword(String keyword) {
```
Goof job on avoiding complicated expressions

wjacobw commented on src/main/java/ChatBot.java:
```diff
@@ -92,6 +96,7 @@ public void deleteTaskByBot(String inputStr) throws deleteException {
      * @throws DukeException If there is an error adding the task
      */
     public void addTaskByBot(String taskType, String description) throws DukeException {
```
good job in making clear and prominent path

wjacobw reviewed (commented):
LGTM!

pangyyen committed:
Rectify test cases after JavaFX

pangyyen committed:
Rectify filepath for release 2.0

pangyyen committed:
Add assertion in Task Packages

Custom exception handle anticipated error scenarios at higher level but don't address
potential programming errors or unintented states internally in the core classes.

Introduce assertions throughout the application core classes. These assertions
will validate the internal state of the objects.

Assertions act as internal self-checks. They are meant to catch programming errors,
not user or external errors.

Relevant Info:
They are typically disabled in a production environment. Therefore, custom
exceptions should continue to be the primary mechanism for handling expected error scenarios.

pangyyen committed:
Add assertion in Task Packages

Custom exception handle anticipated error scenarios at higher level but don't address
potential programming errors or unintented states internally in the core classes.

Introduce assertions throughout the application core classes. These assertions
will validate the internal state of the objects.

Assertions act as internal self-checks. They are meant to catch programming errors,
not user or external errors.

Relevant Info:
They are typically disabled in a production environment. Therefore, custom
exceptions should continue to be the primary mechanism for handling expected error scenarios.

pangyyen committed:
Add assertions in TaskList

pangyyen committed:
Add assertions in TaskList

pangyyen committed:
Refactor UI Initialization and Formatting for Clarity

Previously, the UI setup and formatting process in the `start` method was jumbled
band lacked clear segmentation, making it difficult to trace and manage individual components.

In this commit:
- Extracted UI initialization steps into a separate `initializeGuiComponents`.
- Grouped event handlers in a distinct `setEventHandlers` method to isolate
event-driven logic from the main flow.
- Segregated window formatting logic into a dedicated `formatWindow` method.
- Introduced constant fields for commonly used values.
- Created a `loadResource` utility method, adding an error check for null resources.

With these changes, the UI code now follows a clearer, more modular structure,
making future adjustments and enhancements more straightforward.

pangyyen committed:
Shift the main functionality of Parser to top

pangyyen committed:
Constuct factory methods and introduct Optional type  in Command class

pangyyen committed:
Use Stream to process traversing operation

pangyyen committed:
Merge pull request #1 from pangyyen/branch-A-Assertions

Branch a assertions

pangyyen committed:
Merge updated master to this branch

pangyyen committed:
Merge branch 'master' into branch-A-CodeQuality

master was updated by 'git pull', therefore leaving branch-A-CodeQuality
locally one commit behind

pangyyen committed:
Merge pull request #2 from pangyyen/branch-A-CodeQuality

Branch a code quality

pangyyen committed:
Merge branch 'master' into branch-A-Streams

pangyyen committed:
Merge pull request #3 from pangyyen/branch-A-Streams

Use Stream to process traversing operation

pangyyen committed:
Add gradle.yml for CI

pangyyen committed:
Add gradle.yml for CI

pangyyen committed:
Add gradle.yml for CI

pangyyen committed:
Merge branch 'branch-A-CI'

Add in a gradle.yml for CI

pangyyen committed:
Rectify issue of Ui method comment

Previously, the UI setup and formatting process in the
`start` method was jumbled band lacked clear segmentation,
making it difficult to trace and manage individual components.

In this commit:
- Extracted UI initialization steps into a separate
`initializeGuiComponents`.
- Grouped event handlers in a distinct `setEventHandlers`
method to isolate event-driven logic from the main flow.
- Segregated window formatting logic into a dedicated
`formatWindow` method.
- Introduced constant fields for commonly used values.
- Created a `loadResource` utility method, adding an
error check for null resources.

With these changes, the UI code now follows a clearer,
more modular structure, making future adjustments and
enhancements more straightforward.

pangyyen committed:
Rectify issue of Ui method comment

Previously, the UI setup and formatting process in the
`start` method was jumbled band lacked clear segmentation,
making it difficult to trace and manage individual components.

In this commit:
- Extracted UI initialization steps into a separate
`initializeGuiComponents`.
- Grouped event handlers in a distinct `setEventHandlers`
method to isolate event-driven logic from the main flow.
- Segregated window formatting logic into a dedicated
`formatWindow` method.
- Introduced constant fields for commonly used values.
- Created a `loadResource` utility method, adding an
error check for null resources.

With these changes, the UI code now follows a clearer,
more modular structure, making future adjustments and
enhancements more straightforward.

pangyyen committed:
Merge branch 'branch-A-CodeQuality'

pangyyen committed:
Merge commit 'd04edb4f7a62f6381f55ad03f38a6e3e5871b7e3'

pangyyen committed:
Add command 'help'

pangyyen committed:
Merge branch 'branch-C-Help'

pangyyen committed:
Refactor the location of ParserTest.java

pangyyen committed:
Fix checkstyle

pangyyen committed:
Update README and Ui.png

pangyyen committed:
Update README.md

pangyyen committed:
Update README.md

pangyyen committed:
Move Ui.png and README.md to docs

pangyyen committed:
Improve README

pangyyen committed:
Add README to docs

pangyyen committed:
Update README

pangyyen committed:
Release v2.1.0

pangyyen committed:
Update README
